integration-tests/src/test/java/tachyon/client/RemoteBlockInStreamIntegrationTest.java

```diff
@@ -535,4 +541,42 @@ public void seekAroundLocalBlock() throws IOException, TachyonException {
     Assert.assertEquals(99, is.read());
     is.close();
   }
+
+  /**
+   * Test remote read stream lock in <code>RemoteBlockInStream</code>.
+   */
+  @Test
+  public void remoteReadLockTest() throws Exception {
+    String uniqPath = PathUtils.uniqPath();
+    for (int k = MIN_LEN + DELTA; k <= MAX_LEN; k += DELTA) {
+      TachyonFile f =
+          TachyonFSTestUtils.createByteFile(mTfs, uniqPath + "/file_" + k, k, mWriteTachyon);
+
+      long blockId = mTfs.getInfo(f).getBlockIds().get(0);
+      BlockInfo info = TachyonBlockStore.get().getInfo(blockId);
+
+      NetAddress workerAddr = info.getLocations().get(0).getWorkerAddress();
+      RemoteBlockInStream is =
+          new RemoteBlockInStream(info.getBlockId(), info.getLength(),
+              new InetSocketAddress(workerAddr.getHost(), workerAddr.getDataPort()));
+      byte[] ret = new byte[k / 2];
+      Assert.assertEquals(0, is.read());
+      mTfs.delete(f);
+      CommonUtils.sleepMs(mWorkerToMasterHeartbeatIntervalMs);
```

Comment:
Looks like BlockRemover is a separate thread in thread BlockMasterSync, and it's different with the other heartbeat threads that can use the scheduler, do you think sleeping longer (like 2x) can help here?